### PR TITLE
Reorder queue

### PR DESCRIPTION
**Pull request by tomohung**

### CODE REVIEW

File: app/controllers/queue_items_controller.rb [hunk 1]
```diff
@@ -24,11 +24,26 @@ def create
   def destroy
     queue_item = QueueItem.find(params[:id])
     queue_item.destroy if queue_item.user == current_user
+    current_user.normalize_queue_items
     redirect_to my_queue_path
   end
 
   def update_queue
-    binding.pry
```
Comment (brandoncc) [inline app/controllers/queue_items_controller.rb:31]: Good job cleaning this `pry` up before you committed.

Comment (brandoncc) [inline app/controllers/queue_items_controller.rb:31]: Technically before you pushed :+1:

File: spec/controllers/queue_items_controller_spec.rb [hunk 2]
```diff
@@ -102,6 +102,102 @@
...
       expect(response).to redirect_to sign_in_path      
     end
+
+    it 'normalize the remaining queue items' do
+      user = Fabricate(:user)
+      session[:user_id] = user.id
+      queue_item = Fabricate(:queue_item, user: user, video: Fabricate(:video))
+      queue_item2 = Fabricate(:queue_item, user: user, video: Fabricate(:video))
+      delete :destroy, id: queue_item.id
+      expect(QueueItem.first.position).to eq(queue_item2.reload.position)
```
Comment (brandoncc) [inline spec/controllers/queue_items_controller_spec.rb:112]: This test is not very explicit. I think you should define your position in your fabricators, then use an explicit position number in the expectation too. That way you know for sure it is exactly what you expect.

Comment (brandoncc) [inline spec/controllers/queue_items_controller_spec.rb:112]: Okay, after reading the tests below which are explicit like I mentioned here...it seems that this one is to verify the positioning of items that had no explicit position. I think I would still use an explicit number on the expectation even without the number in the fabricators.

Comment (brandoncc) [inline spec/controllers/queue_items_controller_spec.rb:112]: And....now I just realized that those specs are for a post and this is for a delete. I understand now why you did what you did. Please ignore all three of these comments lol.

Comment (tomohung) [inline spec/controllers/queue_items_controller_spec.rb:112]: As I said, review codes is absolutely NOT an easy job :smile: , thank you!!

**Pull request merged**